### PR TITLE
Add link to the source trigger of the signal notification

### PR DESCRIPTION
**Pull request by spiderjako**
The idea is to link to the source trigger so the user sees what happens

### CODE REVIEW

File: lib/sanbase/signals/signal.ex [hunk 1]
```diff
@@ -39,4 +39,11 @@ defimpl Sanbase.Signal, for: Any do
       {slug, Sanbase.Telegram.send_message(user, payload)}
     end)
   end
+
+  defp extend_payload(user_trigger_id, payload) do
```
Comment (IvanIvanoff) [inline lib/sanbase/signals/signal.ex]: You also need to apply this function where needed

Comment (spiderjako) [reply lib/sanbase/signals/signal.ex]: Yeah, will do it now, thanks :D

File: lib/sanbase/signals/signal.ex [hunk 2]
```diff
@@ -35,8 +35,16 @@ defimpl Sanbase.Signal, for: Any do
       })
       when is_map(payload_map) do
     payload_map
+    |> extend_payload(:user_trigger_id)
```
Comment (IvanIvanoff) [inline lib/sanbase/signals/signal.ex]: this is just passing the atom `:user_trigger_id` and not the actual id

Comment (spiderjako) [reply lib/sanbase/signals/signal.ex]: I've got it now, it should be okay

**Comment by IvanIvanoff** [comment]
@spiderjako can you add a test that checks that new part of the payload?

File: lib/sanbase_web/endpoint.ex [hunk 3]
```diff
@@ -123,4 +123,8 @@ defmodule SanbaseWeb.Endpoint do
       "/verify_email?" <>
       URI.encode_query(token: email_candidate_token, emailCandidate: email_candidate)
   end
+
+  def show_signal_url(id) do
+    sonar_url() <> "/signal/#{id}/edit"
```
Comment (IvanIvanoff) [inline lib/sanbase_web/endpoint.ex]: please update to correct link

Comment (spiderjako) [reply lib/sanbase_web/endpoint.ex]: Alright, it shall be done

File: lib/sanbase/signals/signal.ex [hunk 4]
```diff
@@ -36,7 +37,14 @@ defimpl Sanbase.Signal, for: Any do
       when is_map(payload_map) do
     payload_map
     |> Enum.map(fn {slug, payload} ->
-      {slug, Sanbase.Telegram.send_message(user, payload)}
+      {slug, Sanbase.Telegram.send_message(user, extend_payload(payload, id))}
```
Comment (tspenov) [inline lib/sanbase/signals/signal.ex]: putting it directly in `send` won't put the link in the historical activities: https://app.santiment.net/sonar/activity.
I guess we can start like this and change later if that is requested.

Comment (IvanIvanoff) [reply lib/sanbase/signals/signal.ex]: hm.. Question is do we need it in the historical activity and if we need to have different text?
We can easily extract `extend_payload` and reuse it

Comment (tspenov) [reply lib/sanbase/signals/signal.ex]: I think its is good to link the source trigger there. Though that can be added by frontend with a relative link.

Comment (tspenov) [reply lib/sanbase/signals/signal.ex]: Lets leave it like this for now.

**Pull request merged**